R/issues_dependencies.R

```diff
@@ -0,0 +1,92 @@
+#' @title Report package dependency issues
+#' @export
+#' @family issues
+#' @description Flag packages which have issues in their strong dependencies
+#'   (`Imports:`, `Depends:`, and `LinkingTo:` in the `DESCRIPTION`.)
+#'   These include indirect/upstream dependencies, as well, not just
+#'   the explicit mentions in the `DESCRIPTION` file.
+#' @inheritSection record_issues Package issues
+#' @return A nested list of problems triggered by dependencies.
+#'   The names of top-level elements are packages affected downstream.
+#'   The value of each top-level element is a list whose names are
+#    packages at the source of a problem upstream.
+#'   Each element of this inner list is a character
+#'   vector of relevant dependencies of the downstream package.
+#'
+#'   For example, consider a linear dependency graph where `crew.cluster`
+#'   depends on `crew`, `crew` depends on `mirai`, and
+#'   `mirai` depends on `nanonext`. We represent the graph like this:
+#'      `nanonext -> mirai -> crew -> crew.cluster`.
+#'   If `nanonext` has an issue, then [issues_dependencies()] returns
+#'   `list(crew.cluster = list(nanonext = "crew"), ...)`, where `...`
+#'   stands for additional named list entries. From this list, we deduce
+#'   that `nanonext` is causing an issue affecting `crew.cluster` through
+#'   the direct dependency on `crew`.
+#'
+#'   The choice in output format from [issues_dependencies()] allows package
+#'   maintainers to more easily figure out which direct dependencies
+#'   are contributing issues and drop those direct dependencies if necessary.
+#' @param packages Character vector of names of packages with other issues.
+#' @param meta A data frame with R-universe package check results
+#'   returned by [meta_checks()].
+#' @param verbose `TRUE` to print progress while checking issues with
+#'   dependencies, `FALSE` otherwise.
+#' @examples
+#'   meta <- meta_packages(repo = "https://wlandau.r-universe.dev")
+#'   issues_dependencies(packages = character(0L), meta = meta)
+#'   issues_dependencies(packages = "crew.aws.batch", meta = meta)
+#'   issues_dependencies(packages = "nanonext", meta = meta)
+#'   issues_dependencies(packages = "crew", meta = meta)
+#'   issues_dependencies(packages = c("crew", "mirai"), meta = meta)
+issues_dependencies <- function(
+  packages,
+  meta = multiverse.internals::meta_packages(),
```

Comment:
Here `multiverse.internals::meta_packages()` does not need to be namespaced.
As `meta_packages()` exists in the package namespace, this will always be taken first.

Similarly in the other places you added.

Reply:
done, c.f. 884b1a1206c8c95bad9e5063e1af23283d8d20f6